Gems/Atom/Tools/AtomToolsFramework/Code/Source/EntityPreviewViewport/EntityPreviewViewportScene.cpp

```diff
@@ -74,6 +74,10 @@ namespace AtomToolsFramework
         m_scene->Activate();
 
         AZ::RPI::RPISystemInterface::Get()->RegisterScene(m_scene);
+
+        // set the application-wide MSAA state
+        // Note: this is done after the scene is registered in order to properly update the pipeline
+        AZ::RPI::RPISystemInterface::Get()->SetApplicationMultisampleState(m_activeRenderPipeline->GetRenderSettings().m_multisampleState);
```

Comment:
Shouldn't this still be called from the other function whenever the render pipeline changes at runtime?

Should the call to ActivateRenderPipeline(defaultRenderPipelineAssetPath) be moved down here instead?

Reply:
Looking at this further, I think it's better to go with my original change.  SetApplicationMultisampleState should really only be called once at application startup.

Reply:
It's been a while. I don't recall specific issues related to initialization order, or if the main pipeline needed to be added before brdfTexturePipeline. 

You can change the render pipeline at runtime from a drop down on the viewport in material editor and material canvas. That invokes the activate render pipeline function and updates the application multi sample state for the new pipeline.

If the multi sample state can only be set once on initialization, there should at least be a comment explaining why it was moved out of the activate pipeline function until we can change it dynamically. 

In that case, should it be set to the value from the initial pipeline or to another value that works with all pipelines, even if that's disabled?
Or, should we remove the ability to change the pipeline dynamically in those tools? Ideally, this needs to be fully supported everywhere For visualization and look development.

Reply:
> Looking at this further, I think it's better to go with my original change. SetApplicationMultisampleState should really only be called once at application startup.

OK. But there's an expectation that we can change pipelines at runtime. How will this affect that? Will everything work with the multi sample setting from the default pipeline? Please add comments to the code explaining the change/ impact in case it's revisited. There was a todo in the original code for the same function call but related to supporting multiple viewports.

Reply:
Changing the multisample state at runtime is currently not supported.  You can change pipelines, but changing the number of samples requires a restart.

Reply:
Actually when Chris added a drop down menu to switch between pipeline (in Material editor) it didn't crash when switching between pipelines with msaa and non-msaa. It would be nice to keep/maintain this functionality. Ideally we would want this functionality in the Editor as Carbonated onboards they will want to visualize the look of mobile (i.e mobile non msaa pipeline) in the editor itself. Switching pipelines sounds like a extremely useful feature.

Reply:
what happens if you leave the set msaa state as it currently was, i.e after activating a new pipeline?

Reply:
I can move the ActivateRenderPipeline call for now to resolve this issue, but there's still the possibility of a TDR when changing the MSAA state between pipelines.